server/service/src/invoice/update_picked_date.rs

```diff
@@ -0,0 +1,25 @@
+use chrono::Utc;
+use repository::{
+    InvoiceRow, InvoiceRowRepository, InvoiceStatus, RepositoryError, StorageConnection,
+};
+
+/// This function is called when a line is updated on an invoice. It will update the picked date if appropriate.
+pub fn update_picked_date(
+    connection: &StorageConnection,
+    invoice: &InvoiceRow,
+) -> Result<(), RepositoryError> {
+    // We only want to update the picked date if the invoice is in the picked status and is a prescription
+    if invoice.status != InvoiceStatus::Picked {
+        return Ok(());
+    }
+
+    // Use the invoice's backdated datetime if it's set, otherwise set the status to now
+    let status_datetime = invoice.backdated_datetime.unwrap_or(Utc::now().naive_utc());
```

Comment:
I feel like this logic should be in a helper method (just this line would be ok), and used both here and in prescription/update#generate

Reply:
https://github.com/msupply-foundation/open-msupply/pull/5962/commits/ac4bda7a7ddd57ec836bf5c5373f2f664b91bc95